common/src/main/java/org/ergoplatform/appkit/AppkitProvingInterpreter.scala

```diff
@@ -42,9 +43,10 @@ class AppkitProvingInterpreter(
   import Helpers._
 
   val secrets: Seq[SigmaProtocolPrivateInput[_ <: SigmaProtocol[_], _ <: SigmaProtocolCommonInput[_]]] = {
-    val dlogs = JListToIndexedSeq(identityIso[ExtendedSecretKey]).to(secretKeys).map(_.key)
-    val dhts = JListToIndexedSeq(identityIso[DiffieHellmanTupleProverInput]).to(dhtInputs)
-    dlogs ++ dhts
+    val dlogs: IndexedSeq[DLogProverInput] = try JListToIndexedSeq(identityIso[ExtendedSecretKey]).to(secretKeys).map(_.key) catch {case _:Exception =>  IndexedSeq()}
```

Comment:
Why this `try` is necessary?

Reply:
Otherwise `DHTProverSpec` throws a null pointer exception. See below

`java.lang.NullPointerException was thrown.
java.lang.NullPointerException
	at org.ergoplatform.appkit.AppkitProvingInterpreter.$anonfun$secrets$1(AppkitProvingInterpreter.scala:46)
	at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:238)
	at scala.collection.Iterator.foreach(Iterator.scala:941)
	at scala.collection.Iterator.foreach$(Iterator.scala:941)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1429)
	at scala.collection.IterableLike.foreach(IterableLike.scala:74)
	at scala.collection.IterableLike.foreach$(IterableLike.scala:73)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:56)
	at scala.collection.TraversableLike.map(TraversableLike.scala:238)
	at scala.collection.TraversableLike.map$(TraversableLike.scala:231)
	at scala.collection.AbstractTraversable.map(Traversable.scala:108)
	at org.ergoplatform.appkit.AppkitProvingInterpreter.<init>(AppkitProvingInterpreter.scala:46)
	at org.ergoplatform.appkit.impl.ErgoProverBuilderImpl.build(ErgoProverBuilderImpl.java:138)
	at org.ergoplatform.appkit.DHTProverSpec.$anonfun$new$2(DHTProverSpec.scala:50)
	at org.ergoplatform.appkit.FileMockedErgoClient.execute(FileMockedErgoClient.java:68)
	at org.ergoplatform.appkit.DHTProverSpec.$anonfun$new$1(DHTProverSpec.scala:25)
`

Line 46 is 
`val dlogs: IndexedSeq[DLogProverInput] = JListToIndexedSeq(identityIso[ExtendedSecretKey]).to(secretKeys).map(_.key)`

Reply:
Yes, but why NPE then? What is the root cause? It is probably some bug or misuse.

Reply:
Will investigate and update.

Reply:
Added a null check for _masterKey that was causing this. Essentially, when the prover builder was initialized without master secret, it was giving this error.

Reply:
Made the use of key like the other parameters:

        ArrayList<ExtendedSecretKey> keys = new ArrayList<>();
        if (_masterKey != null) keys.add(_masterKey);

Reply:
NPE occurs because the wallet is not initialized when the prover is created. The above change sets the initial wallet keys similar to how additional keys are set (i.e., with a null check). 
Ideally wallet need not be initialized in many scripts, for example `sigmaProp(1 < 2)`